pkg/filter/filter.go

```diff
@@ -0,0 +1,68 @@
+package filter
+
+import (
+	"net/http"
+)
+
+// Request contains the original http.Request, path parameters and the raw body
+// Request.Request.Body should not be used as it would be already processed by internal implementation
+type Request struct {
+	// Request is the original http.Request
+	*http.Request
+
+	// PathParams contains the URL path parameters
+	PathParams map[string]string
```

Comment:
This should probably be read-only. Maybe an issue can be opened for improvement.

Reply:
why would a path param be changed?